### ISSUE TITLE
Put to sleep a few times and then M keys don't work.

### ISSUE THREAD

**Issue by RamseyKal** [seerge/g-helper#3510]
### Rules

- [X] I made myself familiar with the <a href='https://github.com/seerge/g-helper?tab=readme-ov-file'>Readme</a>, <a href='https://github.com/seerge/g-helper/wiki/FAQ'>FAQ</a> and <a href='https://github.com/seerge/g-helper/wiki/Troubleshooting'>Troubleshooting</a>.
- [X] I understand that, if insufficient information or no app logs will be provided, my issue will be closed without an answer.

### What's wrong?

After I put my laptop to sleep a few times, let's say I use it, put to sleep, use it again, put to sleep, and then I return to use it, the M keys no longer work. Nothing happens when tapping all of them, even after opening G-Helper. Restarting computer doesn't work. Shutting it down and rebooting works. A shut down is required.
I understand that this issue may be related to firmware level and may not be able to be fixed but it's worth checking on it.

### How to reproduce the bug?

Put to sleep a few times.
Then, out of nowhere, when waking up, the M keys no longer work.

### Logs

[log.txt](https://github.com/user-attachments/files/18223297/log.txt)

### Device and Model

Asus ROG Strix G18 G814JZ

### Armoury Crate

Uninstalled

### Asus Services

0

### Version

0.199

### OS

Windows 10 64-Bit 22H2

**Comment by seerge** [comment]
@RamseyKal can you try to simply restart G-Helper when this happens, do keys start working then?

**Comment by seerge** [comment]
@RamseyKal after re-reading your description, I have realized that if laptop restart doesn't help (which includes all apps restart obviously), most likely that won't help too :)

Do M1 and M2 also stop working ? Cause as we have discussed with you before, they just just work as a media Volume keys on firmware level (so you don't need any apps running for them to work).

If so - most likely it's indeed some firmware hiccup. Like it goes to sleep with a laptop, but doesn't wake up after.

**Comment by RamseyKal** [comment]
That's correct, M1 through M5 do not work. None of them. The F keys work fine.
At some point, from experience, if I keep putting it to sleep and waking up after the initial bug, then all the keyboard will stop working until shutting it down. Asus has really buggy laptops compared to Lenovo (I miss those machines but they don't offer 17/18 inch laptops and I don't like small laptops).

**Comment by seerge** [comment]
@RamseyKal I agree :) 

It looks like (another) windows sleep issue that is triggering another firmware hiccup. So keyboard don't completely wake up from sleep.

But unfortunately I can't do anything here then.

**Comment by seerge** [comment]
@RamseyKal no, i'm not aware if that's possible. 

When you start G-Helper it sends some init commands for RGB / keyboard, but since even complete windows restart doesn't change anything - most likely those are not related :)

Since you don't use Modern Standby, it could be that Asus simply never tested / prepared device to work in old-fashioned sleep ...

**Comment by RamseyKal** [comment]
Hi @seerge - I want to reopen this to add more information.

Yesterday and today this issue happened but not because of laptop being put to sleep but this time by closing the lid and re-opening it. It may require me to do it a few times and leaving the laptop idle when lid is closed, and then eventually the M keys don't work.

I am attaching the logs in 2 files.
First file, log.txt, displays the logs at the end of it right before I restarted.

After restart + shut down, I copied the rest of the new logs to another file titled log rest.

The thing is, this issue isn't fixed by just restarting, it requires a shut down and then the M keys are restored.

Also, do you see that error? "Unable to load DLL 'atiadlxx.dll' or one of its dependencies: The specified module could not be found. (0x8007007E)"

No idea what it is.
Do you think it's solvable by G-Helper or it's still an issue by Asus for the G18 2023.

[log.txt](https://github.com/user-attachments/files/19741447/log.txt)

[log rest.txt](https://github.com/user-attachments/files/19741449/log.rest.txt)

**Comment by seerge** [comment]
@RamseyKal hello, if hotkeys stop working after closing/opening lid few times and don't start working again neither after restarting G-Helper (then it would try to reconnect to keyboard to listen for notifications) nor even after reboot - it looks like some strange firmware hiccup.

> "Unable to load DLL 'atiadlxx.dll' or one of its dependencies: The specified module could not be found. (0x8007007E)"

Is just a notification from AMD library that it's not there, since you have Intel device :) it's normal and doesn't affect anything.

**Comment by seerge** [comment]
By the way, keyboard is always available for connection (as USB device), in a log it looks like
```
2025-04-12 9:20:42 PM: Input available: \\?\hid#vid_0b05&pid_19b6&col04#6&15ae9f2&0&0003#{4d1e55b2-f16f-11cf-88cb-001111000030} 19B6 64
2025-04-12 9:20:42 PM: Input: \\?\hid#vid_0b05&pid_19b6&col04#6&15ae9f2&0&0003#{4d1e55b2-f16f-11cf-88cb-001111000030} <--- app is listening for hotkeys input
```
